datalad/distribution/tests/test_publish.py

```diff
@@ -268,24 +293,22 @@ def test_publish_with_data(origin, src_path, dst_path, sub1_pub, sub2_pub, dst_c
     source.repo.add_remote("target", dst_path)
 
     # now, set up targets for the submodules:
-    sub1_target = GitRepo(sub1_pub, create=True)
+    # the need to be annexes, because we want to be able to copy data to them
+    # further down
+    sub1_target = AnnexRepo(sub1_pub, create=True)
```

Comment:
I think that was on purpose -- is should be ok to have superdataset to be a pure git, but subdatasets - annex. and publish the whole collection